api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java

```diff
@@ -677,6 +690,68 @@ public void testGetWorkspace_featuredCategory() {
         .isEqualTo(FeaturedWorkspaceCategory.TUTORIAL_WORKSPACES);
   }
 
+  @Test
+  public void testPublishAlreadyPublishedWorkspace() {
+    // Arrange
+    DbWorkspace dbWorkspace =
+        buildDbWorkspace(
+            workspaceIdIncrementer.getAndIncrement(),
+            "Controlled Tier Workspace",
+            DEFAULT_WORKSPACE_NAMESPACE,
+            WorkspaceActiveStatus.ACTIVE);
+
+    DbFeaturedWorkspace mockDBFeaturedWorkspace =
+        new DbFeaturedWorkspace()
+            .setWorkspace(dbWorkspace)
+            .setCategory(DbFeaturedWorkspace.DbFeaturedCategory.TUTORIAL_WORKSPACES);
+
+    when(mockFeaturedWorkspaceDao.findByWorkspace(dbWorkspace))
+        .thenReturn(Optional.of(mockDBFeaturedWorkspace));
+
+    // Assert
+    assertThrows(
+        BadRequestException.class,
+        () -> workspaceService.publishCommunityWorkspace(dbWorkspace),
+        "Workspace is already featured");
```

Comment:
shouldn't this be "published" ?

Reply:
Why is this passing, i'll check

Reply:
assertThrows doesnt confirm the message for exception as 

 _@Rule and ExpectedException are part of JUnit 4, but when you annotate your test with @org.junit.jupiter.api.Test then the test will be executed by JUnit 5 and JUnit 5 doesn't know how to process @Rule annotations_
 
 Workaround to get the exception object and then check the message